UI/ControlMappingScreen.cpp

```diff
@@ -349,30 +289,148 @@ bool KeyMappingNewKeyDialog::key(const KeyInput &key) {
 	return true;
 }
 
-void KeyMappingNewMouseKeyDialog::CreatePopupContents(UI::ViewGroup *parent) {
+class BindingChoice : public UI::Choice {
+private:
+	std::string bindingText_;
+	bool binding_ = false;
+
+public:
+	BindingChoice(const std::string &text, const std::string& bindingText, UI::LayoutParams *layoutParams = nullptr)
+		: Choice(text, layoutParams), bindingText_(bindingText) {}
+
+	void Draw(UIContext &dc) override;
+	void SetBinding(bool b) { binding_ = b; }
+};
+
+void BindingChoice::Draw(UIContext &dc) {
+	using namespace UI;
+	
+	Style style = dc.theme->itemStyle;
+	if (highlighted_) {
+		style = dc.theme->itemHighlightedStyle;
+	}
+	if (down_) {
+		style = dc.theme->itemDownStyle;
+	}
+	if (HasFocus()) {
+		style = dc.theme->itemFocusedStyle;
+	}
+	if (binding_) {
+		style = dc.theme->itemStyle;
+	}
+	DrawBG(dc, style);
+	
+	if (binding_) {
+		float opacity = sin(5.0*time_now_d())*0.5 + 0.5;
+		uint32_t col = colorAlpha(dc.theme->itemFocusedStyle.background.color, opacity);
+		dc.FillRect(Drawable(col), bounds_);
+	}
+
+	style = dc.theme->itemStyle;
+	dc.SetFontStyle(dc.theme->uiFont);
+
+	const int paddingX = 12;
+	const float availWidth = bounds_.w - paddingX * 2 - textPadding_.horiz();
+	float scale = CalculateTextScale(dc, availWidth);
+
+	dc.SetFontScale(scale, scale);
+
+	Bounds textBounds(bounds_.x + paddingX + textPadding_.left, bounds_.y, availWidth, bounds_.h);
+
+	dc.DrawTextRect(binding_ ? bindingText_.c_str() : text_.c_str(), textBounds, style.fgColor, ALIGN_VCENTER | FLAG_WRAP_TEXT);
+	dc.SetFontScale(1.0f, 1.0f);
+}
+
+void KeyMappingDialog::CreatePopupContents(UI::ViewGroup *parent) {
 	using namespace UI;
 
 	auto km = GetI18NCategory("KeyMapping");
+	auto mc = GetI18NCategory("MappableControls");
+	auto di = GetI18NCategory("Dialog");
+
+	ScrollView *scroll = new ScrollView(ORIENT_VERTICAL, new LinearLayoutParams(FILL_PARENT, 45, 1.0f));
+	LinearLayout *items = new LinearLayoutList(ORIENT_VERTICAL);
+	items->SetSpacing(5.0f);
+
+	static const std::string hint = "Hint: you may map the same key to different PSP button to be able to press multiple one with a single key!";
+	parent->Add(new TextView(km->T(hint), FLAG_DYNAMIC_ASCII | FLAG_WRAP_TEXT, true, new LinearLayoutParams(Margins(10,0))));
```

Comment:
This isn't dynamic ASCII, it can be translated.  That said, I'm worried this might be a bit confusing... not sure if there's a better way.

If we keep it, trying to think of a shorter message... maybe "Hint: Mapping the same key to multiple PSP buttons will press both at once"?

-[Unknown]